diff a/http_parser/pyparser.py b/http_parser/pyparser.py
--- a/http_parser/pyparser.py
+++ b/http_parser/pyparser.py
@@ -306,7 +306,7 @@ class HttpParser(object):
         self._version = (int(match.group(1)), int(match.group(2)))

         # update environ
-        self.environ.update({
+        self._environ.update({
             "PATH_INFO": self._path,
             "QUERY_STRING": self._query_string,
             "RAW_URI": self._url,
